REXLaunchpad referral integration #103

gitChimp88 opened the pull request:
Started creation of REXLaunchpad.test.ts in order to test upcoming integration of referral in to ricochet launchpad.

Fixes: #103

gitChimp88 committed:
Create REXLaunchpad.test.ts

Started creation of REXLaunchpad.test.ts in order to test upcoming integration of referral in to ricochet launchpad

gitChimp88 committed:
Merge branch 'v2' into referral-integration-103

gitChimp88 committed:
start integration of referral to launchpad

gitChimp88 committed:
Fix compilation error

now referencing host properly

gitChimp88 committed:
Update launchpad test

Changed title of test

gitChimp88 committed:
Work out referral fee for affiliate and admin/finish testing

mikeghen commented:
This is looking good, do the tests pass? Is it ready for review?

gitChimp88 commented:
> This is looking good, do the tests pass? Is it ready for review?

@mikeghen Hey! yeah the test passes and it's ready for review.

mikeghen commented on test/REXLaunchpad.test.ts:
```diff
@@ -0,0 +1,440 @@
+            ).to.equal(`true,false,10000,0`);
+            expect(
+                (await launchpad.getIDAShares(ETHX_SUBSCRIPTION_INDEX, bobSigner.address)).toString()
+            ).to.equal(`true,false,10000,0`);
```
Can you add to this, close the stream and add expects for: 
- confirming the affiliate shares go to 0 on stream close
- open a stream without a referral and confirm it doesn't have an affiliate (DAO gets all the feeshares)

gitChimp88 replied:
@mikeghen no problem, changes made!

gitChimp88 committed:
Update REXLaunchpad.test.ts

stopping flow and making sure affiliate shares are 0, creating flow without affilite checking dao gets all shares
